packages/@core.keystone/plugins/softDeleted.js

```diff
@@ -78,11 +76,26 @@ const softDeleted = ({ deletedAtField = 'deletedAt', newIdField = 'newId' } = {}
     return { fields, hooks, access: newAccess, ...rest }
 }
 
-function applySoftDeletedFilters (access, deletedAtField) {
+function getWhereVariables (args) {
+    // TODO(pahaz): this work only for GET query parametrs. Check Post data
+    const variables = get(args, ['context', 'req', 'query', 'variables'])
+    try {
+        return JSON.parse(variables).where || {}
+    } catch (e) {
+        return {}
+    }
+}
+
+function applySoftDeletedFilters (access, deletedAtField, args) {
     const type = getType(access)
     if (type === 'Boolean') {
         return (access) ? { [deletedAtField]: null } : false
     } else if (type === 'Object') {
+        const currentWhereFilters = getWhereVariables(args)
+        if (Object.keys(currentWhereFilters).find((x) => x.startsWith(deletedAtField))) {
```

Comment:
you need a note for that. it's too tricky

Reply:
Yep, noted the hell out of it 🔥

Reply:
we also have a where arguments like this: `where: { OR: [ {deletedAt_not: null}, {deletedAt: null}] }`
Could you please also add a test for that

Reply:
This test broke everything :) Thank you

Problem:
We are only able to check the first level of the `where {...}` object:

```
{where: { deletedAt_not: null } } // OK
{where: { property: { deletedAt_not: null } } } // NOT OK
where: { OR: [ {deletedAt_not: null}, {deletedAt: null}] } // NOT OK
```

Chosen solution: 
1. Deep check if the query contains 'deletedAt' in any place

Other possible solutions:
1. Simply check first N levels - No, because it would be super non intuitive

Implementation caveats and issues:
1. It is possible that sending really deep where queries will be long. Do not know if we should fix this on algorithm side.
2. I didn't find anything like deep check in [lodash](https://lodash.com/docs/4.17.15#isPlainObject) which we use. So I implemented the recursive search myself. Tests are here as well, but my fingers are not as trusted source as _ devs :(